src/CreeDictionary/API/search/eip.py

```diff
@@ -0,0 +1,237 @@
+        for r in self.search_run.unsorted_results():
+            if "+V" in self.new_tags and r.is_lemma and r.lemma_wordform.pos == "V":
+                words.append(r)
+            if "+N" in self.new_tags and r.is_lemma and r.lemma_wordform.pos == "N":
+                words.append(r)
+
+        orig_tags_starting_with_plus: list[str] = []
+        tags_ending_with_plus: list[str] = []
+        for t in self.new_tags:
+            (
+                orig_tags_starting_with_plus
+                if t.startswith("+")
+                else tags_ending_with_plus
+            ).append(t)
+
+        results = []
+        for word in words:
+            # This is sometimes mutated
+            tags_starting_with_plus = orig_tags_starting_with_plus[:]
+
+            noun_tags = ""
+            if "N" in word.wordform.inflectional_category:
+                noun_tags = get_noun_tags(word.wordform.inflectional_category)
+                if "+N" in tags_starting_with_plus:
+                    tags_starting_with_plus.remove("+N")
+                if "+Der/Dim" in tags_starting_with_plus:
+                    # noun tags need to be repeated in this case
+                    index = tags_starting_with_plus.index("+Der/Dim")
+                    tags_starting_with_plus.insert(index + 1, noun_tags)
+
+            text = (
+                "".join(tags_ending_with_plus)
+                + word.lemma_wordform.text
+                + noun_tags
+                + "".join(tags_starting_with_plus)
+            )
+
+            generated_wordforms = strict_generator().lookup(text)
+            for w in generated_wordforms:
+                results.append(
+                    _EipResult(original_result=word, inflected_text=w, analysis=text)
+                )
+        return results
+
+
+# e.g., " swim +V+AI+Prt+3Pl"
+PHRASE_ANALYSIS_OUTPUT_RE = re.compile(
+    r"""
+                        \s*             # leading blank space(s) from flag diacritics
+                        (?P<query>.*)
+                        \s
+                        (?P<tags>\+[^\ ]+)
+                    """,
+    re.VERBOSE,
+)
+
+
+class PhraseAnalyzedQuery:
+    """A structured object holding pieces of, and info about, a phrase query.
+
+    >>> PhraseAnalyzedQuery("they swam").filtered_query
+    'swim'
+    >>> PhraseAnalyzedQuery("they swam").has_tags
+    True
+    >>> PhraseAnalyzedQuery("they swam").tags
+    ['+V', '+AI', '+Prt', '+3Pl']
+    >>> PhraseAnalyzedQuery("excellent").has_tags
+    False
+    """
+
+    def __init__(self, query: str, add_verbose_message=None):
+        self.query = query
+        self.has_tags = False
+        self.filtered_query = None
+        self.tags = None
+        phrase_analyses: list[str] = [
+            r.decode("UTF-8") for r in eng_phrase_to_crk_features_fst()[query]
+        ]
+
+        if add_verbose_message:
+            add_verbose_message({"phrase_analyses": phrase_analyses})
+
+        if len(phrase_analyses) != 1:
```

Comment:
There must be _exactly one_ phrase analyses?

Reply:
For the most part, there should be zero or one phrase analysis, at least as far as I can foresee. At least for now, any phrase analysis always requires at least two English words, and the combinatorics are such that noun and verb phrase analyses are mutually exclusive. Perhaps there might be some verbal phrase cases, where the ambiguity of the verb might result in multiple analyses, e.g. they set, either as present or preterite. But those are really rare.